src/class_loader_core.cpp

```diff
@@ -445,31 +443,32 @@ void loadLibrary(const std::string & library_path, ClassLoader * loader)
     return;
   }
 
-  Poco::SharedLibrary * library_handle = nullptr;
+  #ifdef WIN32
+  HINSTANCE library_handle;
+  #else
+  void * library_handle = nullptr;
+  #endif
   static std::recursive_mutex loader_mutex;
 
   {
     std::lock_guard<std::recursive_mutex> loader_lock(loader_mutex);
 
-    try {
-      setCurrentlyActiveClassLoader(loader);
-      setCurrentlyLoadingLibraryName(library_path);
-      library_handle = new Poco::SharedLibrary(library_path);
-    } catch (const Poco::LibraryLoadException & e) {
-      setCurrentlyLoadingLibraryName("");
-      setCurrentlyActiveClassLoader(nullptr);
-      throw class_loader::LibraryLoadException(
-              "Could not load library (Poco exception = " + std::string(e.message()) + ")");
-    } catch (const Poco::LibraryAlreadyLoadedException & e) {
+    setCurrentlyActiveClassLoader(loader);
+    setCurrentlyLoadingLibraryName(library_path);
+    #ifdef WIN32
+    library_handle = LoadLibrary(library_path.c_str());
+    #else
+    library_handle = dlopen(library_path.c_str(), RTLD_LAZY);
```

Comment:
It would be better to factor this out into a function and perhaps have a wrapping struct which contains either the `void *` or `HINSTANCE` as needed, so that the return type is always the same. And then have functions which operate on that struct to get symbols and things like this. Basically the user (in this case `class_loader`) for these functions should not care what the type of the handle is.